testgrid/cmd/transfigure/cmd/transfigure.go

```diff
@@ -0,0 +1,362 @@
+	cmd := exec.Command("bazel", "test", TestInfraConfigTests, TestInfraVerifySpelling)
+	out, err := runCmd(cmd)
+	log.Print("Bazel Test output: " + out)
+	return err
+}
+
+func ensureGitUserAndEmail() error {
+	log.Print("Checking Git Config user and email")
+	if _, err := runCmd(exec.Command("git", "config", "user.name", o.gitUser)); err != nil {
+		return wrapErrorOrNil("Error setting git user name", err)
+	}
+	if _, err := runCmd(exec.Command("git", "config", "user.email", o.gitEmail)); err != nil {
+		return wrapErrorOrNil("Error setting git user email", err)
+	}
+
+	output, err := runCmd(exec.Command("git", "config", "user.name"))
+	if err != nil {
+		return wrapErrorOrNil("Error on git config user.name", err)
+	}
+	if diff := cmp.Diff(output, o.gitUser); diff != "" {
+		return errors.New("Unexpected Git User: (-got +want)\n" + diff)
+	}
+
+	output, err = runCmd(exec.Command("git", "config", "user.email"))
+	if err != nil {
+		return wrapErrorOrNil("Error on git config user.email", err)
+	}
+	if diff := cmp.Diff(output, o.gitEmail); diff != "" {
+		return errors.New("Unexpected Git Email: (-got +want)\n" + diff)
+	}
+	return nil
+}
+
+func gitCommitAndPush() error {
+	o.prTitle = PrTitlePrefix + o.repoSubdir
+	_, err := runCmd(exec.Command("git", "commit", "-m", o.prTitle))
+	if err != nil {
+		return wrapErrorOrNil("Error on git commit", err)
+	}
+
+	log.Print(fmt.Sprintf("Pushing commit to %s/%s:%s...", o.gitUser, o.remoteForkRepo, Branch))
+	pushTarget := fmt.Sprintf("https://%s:%s@github.com/%s/%s", o.gitUser, o.tokenContents, o.gitUser, o.remoteForkRepo)
+	_, err = runCmd(exec.Command("git", "push", "-f", pushTarget, "HEAD:"+Branch))
+	return wrapErrorOrNil("Error on git push", err)
+}
+
+func createPR() error {
+	log.Printf("Creating PR to merge %s:%s into k8s/test-infra:master...", o.gitUser, Branch)
+	_, err := runCmd(exec.Command("/pr-creator",
+		"--github-token-path="+o.githubToken,
+		"--org=\"kubernetes\"",
+		"--repo=\"test-infra\"",
+		"--branch=master",
+		"--title=\""+o.prTitle+"\"",
+		"--match-title=\""+o.prTitle+"\"",
+		"--body=\"Generated by transfigure cmd\"",
+		"--source=\""+o.gitUser+":"+Branch+"\"",
+		"--confirm"))
+	if err == nil {
+		log.Print("PR created successfully!")
+	}
+	return wrapErrorOrNil("Error creating PR", err)
+}
+
+func runCmd(cmd *exec.Cmd) (string, error) {
+	cmd.Dir = o.workingDir
+	log.Printf("Running command: \n%v", cmd)
+	out, err := cmd.CombinedOutput()
+	return strings.TrimSpace(string(out)), err // Remove trailing newline
+}
+
+func populateGitUserAndEmail() error {
+	// Create a new HTTP request.
+	req, err := http.NewRequest("GET", "https://api.github.com/user", nil)
+	if err != nil {
+		return err
+	}
+	// Append the token we just read from file, and send the request.
+	req.Header.Add("Authorization", "token "+o.tokenContents)
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return wrapErrorOrNil("Error sending request", err)
```

Comment:
If you know that err isn't nil, I would use a direct `fmt.Errorf("Error sending GitHub request: %v", err)` call instead.

Reply:
Done